sequencer/src/header.rs

```diff
@@ -233,6 +234,31 @@ impl Header {
     }
 }
 
+// Function that can be called from `Header::new` for interactiong with
+// l1. When `Header::new` becomes async this logic can be moved there.
+async fn async_header_new(instance_state: &NodeState) -> anyhow::Result<L1Snapshot> {
```

Comment:
I don't know if this function should return a result. I think everything that fails should be retried indefinitely, until HotShot cancels the task because the view has timed out. Basically failing to propose when it's our slot is the worst thing that can happen here so there's no point in failing early if we still have time to retry.

Reply:
Ok. recent commits changed this to retry indefinably.